Document/0x05b-Basic-Security_Testing.md

```diff
@@ -41,29 +41,50 @@ Note: On Linux, you need to choose an SDK directory. `/opt`, `/srv`, and `/usr/l
+| Ease of rooting            | Highly dependent on the device.                                                                                                                                                                                                                              | Typically rooted by default.                                                                                                                          |
+| Ease of emulator detection | It's not an emulator, so very easy.                                                                                                                                                                                                                          | Many artefacts will exist, making it easy to detect that the app is running in an emulator.                                                           |
+| Ease of root detection     | Easier to hide root,  as many root detection algorithms check for emulator properties. With Magisk Systemless root it's nearly impossible to detect.                                                                                                         | Emulators will almost always trigger root detection algorithms due to the fact that they are built for testing with many artifacts that can be found. |
+| Hardware interaction       | Easy interaction through bluetooth, NFC, 4G, WIFI, biometrics, camera, GPS, gyroscope, receiving SMS, ...                                                                                                                                                                   | Usually fairly limited, with emulated hardware input (e.g. random GPS coordinates)                                                                    |
+| API Level support          | Depends on the device and the community. Active communities will keep distributing updated versions (e.g. LineageOS), while less popular devices may only receive a few updates. Switching between versions requires flashing the device, a tedious process. | Always supports the latest versions, including beta releases. Emulators containing specific API levels can easily be downloaded and launched.         |
+| Native library support     | Native libraries are usually built for ARM devices, so they will work on a physical device.                                                                                                                                                                  | Some emulators run on x86 CPUs, so they may not be able to run packaged native libraries.                                                             |
+
 ##### Testing on a Real Device
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1226>
+Almost any physical device can be used for testing, but there are a few considerations to be made. First, the device needs to be rootable. This is not always the case, and some devices may only be unlocked once the carrier contract has been terminated.
+
```

Comment:
Are we mixing rooting and unlocking? Sometimes you can have a radiobaseband locked device and root access right?

Reply:
True, I meant an unlocked bootloader. I've clarified this section.